Classes/extra_functions.py

```diff
@@ -93,4 +93,14 @@ def get_rank_id(settings, name_id):
 
 
 def has_role(role_list, role_id):
-    return len([x for x in role_list if x.id == role_id]) > 0
\ No newline at end of file
+    return len([x for x in role_list if x.id == role_id]) > 0
+
+
+# Process the role_ladder into a usable list when called
+def role_id_index(self, bad_role=0):
+    role_id_ladder = []
+    for entry in self.bot.settings["role_ladder"]:
+        if entry["id"] == bad_role:
+            return entry["name"]
+        role_id_ladder.append(entry["id"])
+    return role_id_ladder
```

Comment:
This function should also be split into two as it is doing two separate tasks, where a function is supposed to be limited to one. This also has a few issues like that if you pass in the wrong ID it will not return None but instead it will return all the roles.

Reply:
Fixed in commit